core/server/web/api/v2/admin/routes.js

```diff
@@ -48,7 +48,7 @@ module.exports = function apiRoutes() {
     router.del('/integrations/:id', mw.authAdminApi, http(apiv2.integrations.destroy));
 
     // ## Schedules
-    router.put('/schedules/:resource/:id', mw.authAdminApi, http(apiv2.schedules.publish));
+    router.put('/schedules/:resource/:id', mw.authInternalAdminApi, http(apiv2.schedules.publish));
```

Comment:
Not sure if we only want to limit this endpoint to the "internal auth" :thinking: Think if the endpoint is a part of the Admin API internal auth should be a special case in addition to regular auth, not a substitution. Let's discuss :smiley:

Reply:
Another option I can think of is making existing auth look up a query string parameter looking for a token instead of a header. Then there's no need to introduce any additional middlewares etc.

Reply:
> Another option I can think of is making existing auth look up a query string parameter looking for a token instead of a header

I was thinking along several options but went with the first one I had to unblock and tie all ends first. My main concern about allowing auth with query string in default admin middleware is that opens up all APIs to allow this form of authentication which is far from ideal imo. We want to restrict this authentication only for a specific endpoint where it will be used. 

Another way to do that might be by making existing auth take a flag which is passed only for this endpoint? 🤔

Reply:
The other option that the URL authentication needs to override is the max age allowed for token, which is currently set as `5m` but in case of scheduler we'll not want to cap the limit with `maxAge` as its already happening with expiry

Reply:
@kevinansfield @gargol Let me know if you have any thoughts on how the middleware for scheduler auth(fetching and verifying token from URL instead of header) should be split out from existing adminApiAuth :)

Reply:
Updated naming to be more explicit but still separate auth middleware for allowing urls with token in them to keep it restricted it to scheduler publish atm

Reply:
will keep thinking about this through the weekend :thinking: it seems like an OK approach, but it also "invents" a new way to access an API so needs some more thinking if we want to create this inventinon

Reply:
@gargol Just to summarize, the main difference between the existing admin auth middleware compared to the auth middleware we need now for scheduler URL is -

- Current middleware picks Token from Auth header of a request. Now we want to be able to pick a token from querystring of a URL to allow scheduled URLs to make authenticated requests.
- Current middleware sets a check of maxAge of `5m` for a token from when it was issued. We don't want any maxAge check for scheduler URLs tokens as they are restricted to a very limited duration which might be long time in future using `neverBefore` and `expiresAt` claims around scheduled publish time.

IMO, both the differentiator points, and specially the second one, is something we don't want to do for default admin auth middleware, as it opens up a lot of security holes for admin APIs, and should only do it for specific endpoints like `schedules`. 

We currently control the token generation for scheduler URLs, which gives us control on setting their validity period. To further keep control over where it can be used, we create a separate middleware(`authAdminApiWithUrl`) which we only allow on `schedules` url and allow it to  bypasses `maxAge` option. 

@kevinansfield Will be good to hear your thoughts on this approach as well 😄

Reply:
I agree that we should restrict this type of auth to only the endpoints where we need it so the current approach seems fine. What we could do is make it really specific? eg. `authSchedulerAdminApi`

Reply:
@kevinansfield 🤔 The main idea behind naming `authAdminApiWithUrl` was its explicit in what type of admin api auth this is rather what it is for. I am ok to go either way on this one though as specific isn't a bad option either in current situation, either ways think it can be changed later 🤷‍♂